src/base/settingsstorage.h

```diff
@@ -64,7 +64,7 @@ class SettingsStorage final : public QObject
     template <typename T>
     T loadValue(const QString &key, const T &defaultValue = {}) const
     {
-        if constexpr (std::is_base_of_v<IStringable, T>)
+        if constexpr (Stringable<T>)
```

Comment:
This requirement is really redundant. It isn't required that class has `T(QString)` constructor.

Reply:
Oops... perhaps I thought about opposite thing.

Reply:
Then it doesn't really require `toString()`.

Reply:
So maybe just add `T(QString)` requirement here without changing anything else?
There is already such a requirement implicitly exists, but using c++20 constraints can improve compile time error handling.

Reply:
>Then it doesn't really require toString().

It is used in `defaultValue.toString()`.

Reply:
>So maybe just add T(QString) requirement here without changing anything else?
There is already such a requirement implicitly exists, but using c++20 constraints can improve compile time error handling.

Another way is to have both `concept Stringable` and `class IStringable` and use the concept here. And use the `IStringable` for class inheritance. However this would duplicate some code.

Reply:
> Another way is to have both `concept Stringable` and `class IStringable` and use the concept here. And use the `IStringable` for class inheritance. However this would duplicate some code.

Can you just require `std::is_constructible_v<T, QString>`?

Reply:
>Can you just require std::is_constructible_v<T, QString>?

Probably. But I don't think separation is a good idea. It fragments the idea of 'Stringable'.

Reply:
>You could define Stringable to satisfy both std::is_constructible_v<T, QString> and std::is_base_of_v<IStringable, T>.

OK, I'll look into it. But I'm not optimistic that Ubuntu CI would be able to compile it.

Reply:
>std::is_base_of_v<IStringable, T>

This would not make a good constraint. For example `QHostAddress` satisfy `Stringable` but it isn't based on `IStringable`.
Anyway I opened #19440.